src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,39 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  onClose: (value: undefined) => void;
+  todo: Todo;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  onClose,
+  todo: { userId, id, title, completed },
+}) => {
+  const [user, setUser] = useState<User>();
+  const [loader, setLoader] = useState(false);
+  const onEscape = (event: KeyboardEvent) => {
+    if (event.key === 'Escape') {
+      onClose(undefined);
+    }
+  };
```

Comment:
The naming convention for event handler functions should follow the pattern `handleEventName`. Consider renaming `onEscape` to `handleEscape` to align with the recommended naming conventions.